simulation/simple_sensor_simulator/test/src/sensor_simulation/detection_sensor/test_detection_sensor.hpp

```diff
@@ -0,0 +1,178 @@
+  }
+
+private:
+  auto configureDetectionSensor() -> void
+  {
+    config_.set_entity("ego");
+    config_.set_architecture_type("awf/universe");
+    config_.set_update_duration(0.1);
+    config_.set_object_recognition_delay(0.0);
+    config_.set_object_recognition_ground_truth_delay(0.0);
+    config_.set_pos_noise_stddev(0.0);
+    config_.set_probability_of_lost(0.0);
+    config_.set_detect_all_objects_in_range(true);
+    config_.set_range(100.0);
+    config_.set_random_seed(12345);
+  }
+
+  auto createRosInterface() -> void
+  {
+    detected_objects_publisher_ =
+      node_->create_publisher<DetectedObjectsMsg>("detected_objects_output", 10);
+    ground_truth_objects_publisher_ =
+      node_->create_publisher<TrackedObjectsMsg>("tracked_objects_output", 10);
+
+    detected_objects_subscriber_ = node_->create_subscription<DetectedObjectsMsg>(
+      "detected_objects_output", 10,
+      [this](const DetectedObjectsMsg::SharedPtr msg) { received_msg_ = msg; });
+  }
+
+  auto initializeEntity(
+    EntityStatus & status, const std::string & name, const uint8_t type, const uint8_t subtype,
+    const geometry_msgs::msg::Pose & pose, const geometry_msgs::msg::Vector3 & dimensions) -> void
+  {
+    status.set_name(name);
+
+    EntityType entity_type;
+    entity_type.type = static_cast<traffic_simulator_msgs::EntityType_Enum>(type);
+    EntitySubtype entity_subtype;
+    entity_subtype.value = static_cast<traffic_simulator_msgs::EntitySubtype_Enum>(subtype);
+
+    simulation_interface::toProto(entity_type, *status.mutable_type());
+    simulation_interface::toProto(entity_subtype, *status.mutable_subtype());
+
+    auto new_pose = status.mutable_pose();
+    auto new_position = new_pose->mutable_position();
+    new_position->set_x(pose.position.x);
+    new_position->set_y(pose.position.y);
+    new_position->set_z(pose.position.z);
+
+    auto new_orientation = new_pose->mutable_orientation();
+    new_orientation->set_x(pose.orientation.x);
+    new_orientation->set_y(pose.orientation.y);
+    new_orientation->set_z(pose.orientation.z);
+    new_orientation->set_w(pose.orientation.w);
+
+    auto new_bounding_box = status.mutable_bounding_box();
+    auto new_dimensions = new_bounding_box->mutable_dimensions();
+    new_dimensions->set_x(dimensions.x);
+    new_dimensions->set_y(dimensions.y);
+    new_dimensions->set_z(dimensions.z);
+  }
+
+  auto createPose(double px, double py, double pz, double ox, double oy, double oz, double ow)
+    -> geometry_msgs::msg::Pose
+  {
+    return geometry_msgs::build<geometry_msgs::msg::Pose>()
+      .position(geometry_msgs::build<geometry_msgs::msg::Point>().x(px).y(py).z(pz))
+      .orientation(geometry_msgs::build<geometry_msgs::msg::Quaternion>().x(ox).y(oy).z(oz).w(ow));
+  }
+
+  auto createDimensions(double x, double y, double z) -> geometry_msgs::msg::Vector3
+  {
+    return geometry_msgs::build<geometry_msgs::msg::Vector3>().x(x).y(y).z(z);
+  }
```

Comment:
these functions too. these helper/make functions could be aggregated in one file and not repeated in each class

Reply:
moved to utils/helper_functions